README.md

```diff
@@ -127,9 +169,69 @@ print 'Wrote record {0}'.format(record.meta.record_id)
 
 ## Searching records
 
-E3DB supports complex search options for finding records based on the fields stored in record metadata. 
+E3DB supports complex search options for finding records based on the terms stored in record metadata. 
+These fields include the properties in the [Meta class](e3db/types/meta.py):
+```
+ - record_ids
+ - writer_ids
+ - user_ids
+ - record_types
+ - plain
+    - keys
+    - values
+```
+
+The values in the plain meta dictionary are expanded into keys and values for two additional query fields:
 
-For example, to list all records of type `contact` and print a simple report containing names and phone numbers:
+```
+ - keys
+ - values
+
+WHERE
+
+ plain = {"key1":"value1", "key2":"value2"}
+ keys = ["key1", "key2"]
+ values = ["value1", "value2"]
+```
+
+To further narrow your search, these a range can be provided for these fields:
+```
+ - created
+ - last_modified
+```
+
+#### Search Results
+
+The results of your search come back in a [SearchResult object](e3db/types/search_result.py):
+
+```python
+# This object contains your records,
+# and additional information about your search.
+
+query = Search(include_data=True)
+results = client.search(query)
+
+# Records found in your search
+results.records 
+
+# Number of records returned from search
+len(results) 
```

Comment:
Is there a `total_fields` field we can use more cheaply than iterating over and counting up all the results? You've seen external client code, they can use all the good coding examples we can provide ;-)

Reply:
len is an O(1) operation in python

Reply:
dsda